packages/back-end/src/controllers/stripe.ts

```diff
@@ -4,106 +4,141 @@ import {
 import {
   updateOrganization,
-  updateOrganizationByStripeId,
+  findOrganizationByStripeCustomerId,
 } from "../models/OrganizationModel";
-import { createOrganization } from "../models/OrganizationModel";
 import { getOrgFromReq } from "../services/organizations";
+import { updateSubscription } from "../services/stripe";
 const stripe = new Stripe(STRIPE_SECRET || "", { apiVersion: "2020-08-27" });
 
-async function updateSubscription(subscription: string | Stripe.Subscription) {
-  // Make sure we have the full subscription object
-  if (typeof subscription === "string") {
-    subscription = await stripe.subscriptions.retrieve(subscription);
-  }
+let priceData: Stripe.Price;
 
-  const stripeCustomerId =
-    typeof subscription.customer === "string"
-      ? subscription.customer
-      : subscription.customer.id;
-
-  await updateOrganizationByStripeId(stripeCustomerId, {
-    subscription: {
-      id: subscription.id,
-      qty: subscription.items.data[0].quantity || 1,
-      trialEnd: subscription.trial_end
-        ? new Date(subscription.trial_end * 1000)
-        : null,
-      status: subscription.status,
-    },
-  });
-}
-export async function postStartTrial(
-  req: AuthRequest<{ qty: number; name: string }>,
+export async function postNewSubscription(
+  req: AuthRequest<{ qty: number; restart: boolean }>,
   res: Response
 ) {
-  const { qty, name } = req.body;
-
-  // If user already has a subscription, return immediately
-  if (req.organization?.subscription?.id) {
-    return res.status(200).json({
-      status: 200,
-    });
-  }
+  const { qty, restart } = req.body;
 
   try {
-    // Create organization first if needed
-    if (!req.organization) {
-      if (name.length < 3) {
-        throw new Error("Company name must be at least 3 characters long");
+    req.checkPermissions("organizationSettings");
+
+    const { org } = getOrgFromReq(req);
+
+    if (!org) {
+      throw new Error("No organization found");
+    }
+
+    const qtyInDb = org.members.length + (org.invites?.length || 0);
+
+    if (restart) {
+      if (qty !== qtyInDb) {
+        throw new Error("Error with quantity of users");
+      }
+    } else {
+      if (qty !== qtyInDb + 1) {
+        throw new Error("Error with quantity of users");
       }
-      req.organization = await createOrganization(
-        req.email || "",
-        req.userId || "",
-        name,
-        ""
-      );
     }
 
-    // Create customer in Stripe if not exists
-    if (!req.organization.stripeCustomerId) {
-      const resp = await stripe.customers.create({
-        email: req.email || "",
-        name: req.name || "",
+    let stripeCustomerId: string;
+
+    if (org.stripeCustomerId) {
+      stripeCustomerId = org.stripeCustomerId;
+    } else {
+      const { id } = await stripe.customers.create({
         metadata: {
-          user: req.userId || "",
-          organization: req.organization.id,
+          growthBookId: org.id,
```

Comment:
Might be worth adding the organization name and possibly the ownerEmail here.  Otherwise, when looking in the Stripe dashboard, we'll have no idea who any of the customers belong to.

Reply:
Good call. Added that in.

Reply:
I was able to pass in the org.name outside of the metadata, but I left the org.ownerEmail as metadata. When the user actually starts a subscription, the email will then be populated with the email entered at checkout. If you think passing the ownerEmail in as the customer's email is better, just let me know.